LLPlot.py

```diff
@@ -0,0 +1,176 @@
+
+                    for i in commands[1:]:
+                        if len(commands) <= 1:
+                            continue
+
+                        elif i[0].upper() == 'X':
+                            col['x'] = float(i[1:])
+                            self.__min_max('x', float(i[1:]))
+
+                        elif i[0].upper() == 'Y':
+                            continue
+
+                        elif i[0].upper() == 'Z':
+                            col['z'] = float(i[1:])
+                            self.__min_max('y', float(i[1:]))
+
+                        elif i[0].upper() == 'F':
+                            continue
+
+                        else:
+                            print(line)
+                            raise Warning('Unknown character!')
+                else:
+                    continue
+
+                code.append(col)
+        self.__draw_image(code)
+
+    def __reset_min_max(self):
+        self.__min_x = 0
+        self.__min_y = 0
+        self.__max_x = 0
+        self.__max_y = 0
+
+    def __min_max(self, coordinate, value):
+        # Used to derive the image size based on g code coordinates
+        if coordinate == 'x':
+            self.__min_x = min([value, self.__min_x])
+            self.__max_x = max([value, self.__max_x])
+        else:
+            self.__min_y = min([value, self.__min_y])
+            self.__max_y = max([value, self.__max_y])
+
+    def __image_size(self):
+        # Get the image size
+        x = math.ceil(abs(self.__min_x - self.__max_x))
+        y = math.ceil(abs(self.__min_y - self.__max_y))
+
+        # divide by image size
+        x_scale = math.floor(self.imageSize[0] / x)
+        y_scale = math.floor(self.imageSize[1] / y)
+
+        # scale up
+        self.imageSize = ((x * x_scale) + 50, (y * y_scale) + 50)
+
+        return min([x_scale, y_scale])
+
+    def __draw_image(self, code):
+        # size of the image (should be based on the max path point)
+        scale = self.__image_size()
+        img = Image.new('RGB', self.imageSize, self.background)
+        draw = ImageDraw.Draw(img)
+
+        i = 0
+        x = 1
+        while i <= (len(code) - 1):
+            line_color = self.__get_line_color(code[i]['g'][0])
+
+            # / 2 to draw from the center of the image
+            # - 25 to offset the draw point from the edge of the image
+            x_start = (self.imageSize[0] / 2) + (code[i]['z'] * scale) - 25
+            y_start = self.imageSize[1] + (code[i]['x'] * scale) - 25
+            x_end = (self.imageSize[0] / 2) + (code[x]['z'] * scale) - 25
+            y_end = self.imageSize[1] + (code[x]['x'] * scale) - 25
+
+            draw.line((x_start, y_start, x_end, y_end), fill=line_color, width=self.lineThickness)
+            i += 1
+            if x != (len(code) - 1):
+                x += 1
+
+        # Mirror because its draw flipped.
+        img = ImageOps.flip(img)
+        img.save(self.imageName + self.imageType)
```

Comment:
could you add the option to save to a specified path?

Reply:
Added in commit 0933275